pkg/storage/chunk/cache/redis_client.go

```diff
@@ -51,9 +52,27 @@ type RedisClient struct {
 }
 
 // NewRedisClient creates Redis client
-func NewRedisClient(cfg *RedisConfig) *RedisClient {
+func NewRedisClient(cfg *RedisConfig) (*RedisClient, error) {
```

Comment:
I think would be nice to add few simples tests to lock this behaviour in `NewRedisClient` with different values for `cfg.Endpoint` (single vs multiple comma separated strings?)

We may not need to patch Go resolver during runtime (like you mentioned on the PR description). How about just passing list comma separated IPs and not hostnames? Will that help?

Reply:
We already have 2 tests here which pass a single endpoint and then 2 endpoints as a cluster using mock redis instances: https://github.com/grafana/loki/blob/main/pkg/storage/chunk/cache/redis_client_test.go#L67-L100
I assume these are passing IPs to the client config, so they should already cover what you mentioned.